datahub-classify/src/datahub_classify/infotype_helper.py

```diff
@@ -185,10 +185,10 @@ def inspect_for_gender(
 
     try:
         if (
-            hasattr(debug_info, "name")
-            and int(debug_info.name) == 1
-            and hasattr(debug_info, "values")
-            and debug_info.values == 0
+            prediction_factors_weights.get(NAME, 0) > 0
+            and debug_info.name == 1.0
```

Comment:
Modified the code not to use "==" for float comparison. Also I have modified DebugInfo instance variable type to float instead of float and str. Added couple of TODOs regarding adding warning/error messages  in the error flag which will get passed in ColumnInfo object in future PR.